Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx

```diff
@@ -139,29 +140,30 @@ CastImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateDataDispatche
 
   this->CallCopyOutputRegionToInputRegion(inputRegionForThread, outputRegionForThread);
```

Comment:
For my understanding, do inputRegionForThread and outputRegionForThread always have exactly the same number of pixels (as returned by `ImageRegion::GetNumberOfPixels()`), after calling `CallCopyOutputRegionToInputRegion`?

Reply:
Looking at an older version, it looks like it assumes that the output region has the same size as the input region. Because it just checks inputIt.IsAtEnd() and inputIt.IsAtEndOfLine(), not outputIt.IsAtEnd() or outputIt.IsAtEndOfLine(): https://github.com/InsightSoftwareConsortium/ITK/blob/921c5bab8f0803224d6b7cd7c5ab14aefcfe89ff/Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx#L145-L165

That's OK to me, as long as that assumption is OK  😃 So then, in the new code, it's also OK to just check `inputRange.end()` and ignore `outputRange.end()`.

Reply:
Cast image filter does not need a larger input region, so it does not override the default `CallCopyOutputRegionToInputRegion` implementation, and has the same input and output regions.